src/tests/test_magic.py

```diff
@@ -181,6 +181,54 @@ def test_persist_frame_at_its_creation(ip):
     assert "Shakespeare" in str(persisted)
 
 
```

Comment:
To make sure we are not missing anything, can we also test on different tables (i.e. `number_table`) and on their actual values (and not just the count)? Also, I think it will be easier if we'll use parameters here.

Reply:
> To make sure we are not missing anything, can we also test on different tables (i.e. `number_table`) and on their actual values (and not just the count)? Also, I think it will be easier if we'll use parameters here.

this comment relates to all upcoming 4 tests

Reply:
1. Added different tables as parameters
2. Added more tests 
(When we are using --persist-replace at the beginning (without --persist)
When we are using --persist-replace twice)